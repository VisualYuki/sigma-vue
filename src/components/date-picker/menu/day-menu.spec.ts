import {expect, describe, it, beforeEach, vi} from 'vitest'
import {mount, VueWrapper} from '@vue/test-utils'
import DayMenu from './day-menu.vue'

describe('DayMenu with single select mode', () => {
	let wrapper: VueWrapper<InstanceType<typeof DayMenu>>

	beforeEach(() => {
		vi.useFakeTimers()
		vi.setSystemTime(new Date(2025, 7, 17, 10, 30, 45))

		wrapper = mount(DayMenu)
	})

	it('render snapshot', () => {
		expect(wrapper.html()).toMatchSnapshot()
	})

	describe('when clicking on 15th day', () => {
		beforeEach(() => {
			wrapper.find('[data-day="15"]').trigger('click')
		})

		it('should set selected date to 15th day', () => {
			expect(wrapper.vm.selectedDate?.format('YYYY-MM-DD')).toBe('2025-08-15')
		})

		describe('when clicking on 16th day', () => {
			beforeEach(() => {
				wrapper.find('[data-day="16"]').trigger('click')
			})

			it('should set selected date to 16th day', () => {
				expect(wrapper.vm.selectedDate?.format('YYYY-MM-DD')).toBe('2025-08-16')
			})
		})
	})
})

describe('DayMenu with range select mode', () => {
	let wrapper: VueWrapper<InstanceType<typeof DayMenu>>

	beforeEach(() => {
		vi.useFakeTimers()
		vi.setSystemTime(new Date(2025, 7, 17, 10, 30, 45))

		wrapper = mount(DayMenu, {
			props: {
				selectMode: 'range'
			}
		})
	})

	it('render snapshot', () => {
		expect(wrapper.html()).toMatchSnapshot()
	})

	describe('when clicking on 15th day', () => {
		beforeEach(() => {
			wrapper.find('[data-day="15"]').trigger('click')
		})

		it('should set start selected date to 15th day', () => {
			expect(wrapper.vm.startSelectedDate?.format('YYYY-MM-DD')).toBe('2025-08-15')
		})

		describe('when clicking on 17th day', () => {
			beforeEach(() => {
				wrapper.find('[data-day="17"]').trigger('click')
			})

			it('should set end selected date to 17th day', () => {
				expect(wrapper.vm.endSelectedDate?.format('YYYY-MM-DD')).toBe('2025-08-17')
			})

			it('there is in-range state for 16th day', () => {
				expect(wrapper.find('[data-day="16"]').attributes('data-state')).toBe('in-range')
			})
		})
	})
})
