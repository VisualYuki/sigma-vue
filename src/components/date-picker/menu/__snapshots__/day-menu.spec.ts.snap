// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DayMenu with single select mode > render snapshot 1`] = `
"<div class="p-5">
  <div class="flex gap-1 justify-between items-center text-blue-500"><button class="rounded-md font-medium inline-flex items-center justify-around cursor-pointer select-none disabled-utility h-8 focus-utility p-1 text-sm border border-blue-500 text-blue-500" data-name="ui-button">
      <!---->
    </button>
    <p class="first-letter:uppercase">август 2025</p><button class="rounded-md font-medium inline-flex items-center justify-around cursor-pointer select-none disabled-utility h-8 focus-utility p-1 text-sm border border-blue-500 text-blue-500" data-name="ui-button">
      <!---->
    </button>
  </div>
  <div class="mb-3.5">
    <div class="grid grid-cols-[repeat(7,30px)] gap-1 h-6 mb-2 text-center">
      <div>Пн</div>
      <div>Вт</div>
      <div>Ср</div>
      <div>Чт</div>
      <div>Пт</div>
      <div>Сб</div>
      <div>Вс</div>
    </div>
    <div class="grid grid-cols-[repeat(7,30px)] gap-1">
      <div class="text-gray-500 border border-transparent hover:border-gray-500 rounded-full h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer">28</div>
      <div class="text-gray-500 border border-transparent hover:border-gray-500 rounded-full h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer">29</div>
      <div class="text-gray-500 border border-transparent hover:border-gray-500 rounded-full h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer">30</div>
      <div class="text-gray-500 border border-transparent hover:border-gray-500 rounded-full h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer">31</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="1" data-state="unselected">1</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="2" data-state="unselected">2</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="3" data-state="unselected">3</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="4" data-state="unselected">4</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="5" data-state="unselected">5</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="6" data-state="unselected">6</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="7" data-state="unselected">7</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="8" data-state="unselected">8</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="9" data-state="unselected">9</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="10" data-state="unselected">10</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="11" data-state="unselected">11</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="12" data-state="unselected">12</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="13" data-state="unselected">13</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="14" data-state="unselected">14</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="15" data-state="unselected">15</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="16" data-state="unselected">16</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="17" data-state="unselected">17</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="18" data-state="unselected">18</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="19" data-state="unselected">19</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="20" data-state="unselected">20</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="21" data-state="unselected">21</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="22" data-state="unselected">22</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="23" data-state="unselected">23</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="24" data-state="unselected">24</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="25" data-state="unselected">25</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="26" data-state="unselected">26</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="27" data-state="unselected">27</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="28" data-state="unselected">28</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="29" data-state="unselected">29</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="30" data-state="unselected">30</div>
      <div class="h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer hover:bg-gray-100" data-day="31" data-state="unselected">31</div>
      <div class="text-gray-500 border border-transparent hover:border-gray-500 rounded-full h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer">1</div>
      <div class="text-gray-500 border border-transparent hover:border-gray-500 rounded-full h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer">2</div>
      <div class="text-gray-500 border border-transparent hover:border-gray-500 rounded-full h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer">3</div>
      <div class="text-gray-500 border border-transparent hover:border-gray-500 rounded-full h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer">4</div>
      <div class="text-gray-500 border border-transparent hover:border-gray-500 rounded-full h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer">5</div>
      <div class="text-gray-500 border border-transparent hover:border-gray-500 rounded-full h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer">6</div>
      <div class="text-gray-500 border border-transparent hover:border-gray-500 rounded-full h-7.5 flex justify-center items-center rounded-full select-none cursor-pointer">7</div>
    </div>
  </div>
  <div class=""><button class="rounded-md font-medium inline-flex items-center justify-around cursor-pointer select-none disabled-utility h-8 focus-utility p-1 text-sm border border-blue-500 text-blue-500" data-name="ui-button">Today</button></div>
</div>"
`;
